### ISSUE TITLE
Test that operations equivalently panic

### ISSUE THREAD

**Issue by Nemo157** [jakubadamw/rutenspitz#2]
It would be useful to be able to specify that some methods are expected to sometimes panic, then have the running of those methods be wrapped in `catch_unwind` and the result of that compared to ensure both implementations either panic, or return the same result.

It may or may not be possible to continue running other operations after the methods panic, so these might need to be tagged somehow as to whether the data structure/operation is `RefUnwindSafe`.

**Comment by jakubadamw** [comment]
This is a good idea, thanks!

**Comment by Shnatsel** [comment]
I've [attempted to implement this](https://github.com/Shnatsel/arbitrary-model-tests/commit/a09a044f700f39943fcbdd134ad57f33728a25d4) but ended up in a weird lifetime hell:

```
error[E0495]: cannot infer an appropriate lifetime for autoref due to conflicting requirements
   --> src/tests/index_map.rs:168:16
    |
168 |             fn values_mut(&mut self) -> impl Iterator<Item = &mut V>;
    |                ^^^^^^^^^^
    |
note: first, the lifetime cannot outlive the lifetime `'_` as defined on the body at 135:1...
   --> src/tests/index_map.rs:135:1
    |
135 |   arbitrary_stateful_operations! {
    |  _-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
136 | |     model = ModelHashMap<K, V>,
137 | |     tested = IndexMap<K, V>,
138 | |
...   |
170 | |     }
171 | | }
    | |_- in this macro invocation
note: ...so that closure can access `model`
   --> src/tests/index_map.rs:135:1
    |
135 | / arbitrary_stateful_operations! {
136 | |     model = ModelHashMap<K, V>,
137 | |     tested = IndexMap<K, V>,
138 | |
...   |
170 | |     }
171 | | }
    | | ^ in this macro invocation
    | |_|
    | 
note: but, the lifetime must be valid for the call at 135:1...
   --> src/tests/index_map.rs:135:1
    |
135 | / arbitrary_stateful_operations! {
136 | |     model = ModelHashMap<K, V>,
137 | |     tested = IndexMap<K, V>,
138 | |
...   |
170 | |     }
171 | | }
    | | ^ in this macro invocation
    | |_|
    | 
note: ...so type `std::result::Result<impl std::iter::Iterator, std::boxed::Box<(dyn std::any::Any + std::marker::Send + 'static)>>` of expression is valid during the expression
   --> src/tests/index_map.rs:135:1
    |
135 | / arbitrary_stateful_operations! {
136 | |     model = ModelHashMap<K, V>,
137 | |     tested = IndexMap<K, V>,
138 | |
...   |
170 | |     }
171 | | }
    | | ^ in this macro invocation
    | |_|
    | 
```

**Comment by jakubadamw** [comment]
Closed by #7.
